driver/ppm_fillers.c

```diff
@@ -884,7 +896,7 @@ static bool ppm_is_upper_layer(struct file *file)
 		struct ovl_entry *oe = (struct ovl_entry*)(dentry->d_fsdata);
 		if(!oe)
 		{
-			return false;
+			return PPM_OVERLAY_LOWER;
```

Comment:
WDYT? Same for below points

```suggestion
			return PPM_NOT_OVERLAY_FS;
```